Lib/enum.py

```diff
@@ -481,18 +482,149 @@ def __new__(metacls, cls, bases, classdict, *, boundary=None, _simple=False, **k
+        classdict.update(enum_class.__dict__)
+        #
+        # create a default docstring if one has not been provided
+        if enum_class.__doc__ is None:
+            if not member_names:
+                enum_class.__doc__ = classdict['__doc__'] = _dedent("""\
+                        Create a collection of name/value pairs.
+
+                        Example enumeration:
+
+                        >>> class Color(Enum):
+                        ...     RED = 1
+                        ...     BLUE = 2
+                        ...     GREEN = 3
+
+                        Access them by:
+
+                        - attribute access::
+
+                        >>> Color.RED
+                        <Color.RED: 1>
+
+                        - value lookup:
+
+                        >>> Color(1)
+                        <Color.RED: 1>
+
+                        - name lookup:
+
+                        >>> Color['RED']
+                        <Color.RED: 1>
+
+                        Enumerations can be iterated over, and know how many members they have:
+
+                        >>> len(Color)
+                        3
+
+                        >>> list(Color)
+                        [<Color.RED: 1>, <Color.BLUE: 2>, <Color.GREEN: 3>]
+
+                        Methods can be added to enumerations, and members can have their own
+                        attributes -- see the documentation for details.
+                        """)
+            else:
+                member = list(enum_class)[0]
+                enum_length = len(enum_class)
+                cls_name = enum_class.__name__
+                if enum_length == 1:
+                    list_line = 'list(%s)' % cls_name
+                    list_repr = '[<%s.%s: %r>]' % (cls_name, member.name, member.value)
+                elif enum_length == 2:
+                    member2 = list(enum_class)[1]
+                    list_line = 'list(%s)' % cls_name
+                    list_repr = '[<%s.%s: %r>, <%s.%s: %r>]' % (
+                            cls_name, member.name, member.value,
+                            cls_name, member2.name, member2.value,
+                            )
+                elif enum_length == 3:
+                    member2 = list(enum_class)[1]
+                    member3 = list(enum_class)[2]
+                    list_line = 'list(%s)' % cls_name
+                    list_repr = '[<%s.%s: %r>, <%s.%s: %r>, <%s.%s: %r>]' % (
+                            cls_name, member.name, member.value,
+                            cls_name, member2.name, member2.value,
+                            cls_name, member3.name, member3.value,
+                            )
+                else:
+                    member2 = list(enum_class)[1]
+                    member3 = list(enum_class)[2]
+                    list_line = 'list(%s)[:3]' % cls_name
+                    list_repr = '[<%s.%s: %r>, <%s.%s: %r>, <%s.%s: %r>]' % (
```

Comment:
Probably `enum_length == 3` and `else` can be unified.

Reply:
Done.